Add missing options of the copy and incremental import commands (#2068)
neo4j/docs-operations: modules/ROOT/pages/backup-restore/copy-database.adoc

[source,role=noheader]
----
neo4j-admin database copy [-h] [--copy-schema] [--expand-commands] [--force] [--verbose] [--compact-node-store[=true|false]]

>                              [--additional-config=<file>] [--from-pagecache] [--max-off-heap-memory=<size>]

Author: ```suggestion
                             [--additional-config=<file>] [--from-pagecache=<size>] [--from-pagecache, --max-off-heap-memory=<size>]
```

|Force the command to run even if the integrity of the database cannot be verified.
|

> |--from-pagecache

Author: ```suggestion
|--from-pagecache=<size>
```

Cannot be combined with `--copy-only-relationships-with-types`.
|

> |--max-off-heap-memory=<size>

Author: ```suggestion
|--from-pagecache, --max-off-heap-memory=<size>
```

> |Maximum memory that neo4j-admin can use for various data structures and caching to improve performance.

Author: ```suggestion
|label:new[Introduced in 2025.01] Maximum memory that neo4j-admin can use for various data structures and caching to improve performance.
```

Values can be plain numbers, such as 10000000, or 20G for 20 gigabytes.

> It can also be specified as a percentage of the available memory, for example 70%%.

Author: ```suggestion
It can also be specified as a percentage of the available memory, for example 70%.
```

> |

Author: ```suggestion
|90%
```

For more information on the block format, see xref:database-internals/store-formats.adoc[Store formats].
====

[NOTE]
====
In addition to the `--from-pagecache=<size>` option, a new option `--max-off-heap-memory` is introduced in Neo4j 2025.01.

> You can still use the `--from-pagecache` option to speed up the copy operation by specifying how much cache to allocate when reading the source.

Reviewer: `--from-pagecache` controls the same value as `--max-off-heap-memory`, therefor it's no longer about how much cache to allocate when reading the source.

Reviewer 2: @tinwelint, we say that `--max-off-heap-memory` is added in addition to the `--from-pagecache=<size>`, but it actually replaces it, which means `--from-pagecache=<size>` is removed, not deprecated. I am not sure I understand the process here.

Reviewer 2: If it is deprecated, then it should be available both in the syntax and as an argument, no?

Reviewer: The function of `--from-pagecache` changed and it got another more appropriate name too - this to not break scripts that already used `--from-pagecache`. I don't think we can tell PicoCLI (the lib we use to parse/run commands) to deprecate a certain option name.

Reviewer 2: So, it is not deprecated. It is enhanced 😄. Thanks!

For more information on the block format, see xref:database-internals/store-formats.adoc[Store formats].
====

[NOTE]
====

> In Neo4j 2025.01, a new dual-named option `--from-pagecache, --max-off-heap-memory=<size>` replaces the `--from-pagecache=<size>` option, which was used to specify how much cache to allocate when reading the source.

Author: I'm not sure that 'enhance' is a correct word here. Previously, you could only allocate cache, but now you govern off-heap memory. I might be wrong but these seem to be two separate tasks. So, you don't enhance the functionality of the `--from-pagecache=<size>` option, but replace it with a new one.

> In Neo4j 2025.01, a new dual-named option `--from-pagecache, --max-off-heap-memory=<size>` replaces the `--from-pagecache=<size>` option, which was used to specify how much cache to allocate when reading the source.

Reviewer 2: So, it doesn't allocate cache. I was confused by the sentence, "This change helps you to control off-heap memory rather than just allocate cache."

> In Neo4j 2025.01, a new dual-named option `--from-pagecache, --max-off-heap-memory=<size>` replaces the `--from-pagecache=<size>` option, which was used to specify how much cache to allocate when reading the source.

Author: I revisited the original PR in the code and reread once again everything related to this change. I'll push some changes where some of your suggestions, @renetapopova, are implemented.

> The new option determines the maximum amount of off-heap memory available during the copy process for reading and writing.

Reviewer 2: ```suggestion
The new option determines the maximum amount of off-heap memory available during the copy process for reading and writing,  instead of specifying how much cache to allocate when reading the source.
```

> This change helps you to control off-heap memory rather than just allocate cache.

Reviewer 2: ```suggestion
```

> To configure the off-heap memory value, you can use either `--from-pagecache=<size>` or `--max-off-heap-memory=<size>`.

Reviewer 2: This sentence claims that both are available, but we don't have descriptions and values for both, and we don't explain when to use one or another or both together.

Author: Yes, that's correct, they are both available. We have one description and one default value for both.

if I understood the idea correctly, you still can use the old `--from-pagecache=<size>`, but now it will control the off-heap memory value, not the cache allocation and its default value now is `90%` not `8m`.

You cannot use them together. You have to use either  the old name`--from-pagecache=<size>` or the new name `--max-off-heap-memory=<size>`.

Reviewer 2: > Yes, that's correct, they are both available. We have one description and one default value for both. In other words, they share one description and one default value.
> 
> if I understood the idea correctly, you still can use the old `--from-pagecache=<size>`, but now it will control the off-heap memory value, not the cache allocation and its default value now is `90%` not `8m`.
> 
> You cannot use them together. You have to use either the old name`--from-pagecache=<size>` or the new name `--max-off-heap-memory=<size>`.

If that's true, then we need. to change this sentence to say it:

> To configure the off-heap memory value, you can use either `--from-pagecache=<size>` or `--max-off-heap-memory=<size>`.

Reviewer 2: ```suggestion
To configure the off-heap memory value, you can use either `--from-pagecache=<size>` or `To configure the off-heap memory value, you can use either --from-pagecache=<size> or --from-pagecache --max-off-heap-memory=<size> --max-off-heap-memory=<size>`.
```

> Note that the `--from-pagecache` option is considered deprecated and may be removed in future versions.

Reviewer 2: Is it deprecated or not?

Author: I guess the old name `--from-pagecache` will be removed in the future, so I suppose we have to say about it. If the word 'deprecated' is too confusing, we need to rephrase this sentence. 
@tinwelint, any comments?

> Note that the `--from-pagecache` option is considered deprecated and may be removed in future versions.

Author: ```suggestion
Note that the `--from-pagecache` option is considered outdated and may be removed in future versions.
```

> Note that the `--from-pagecache` option is considered deprecated and may be removed in future versions.

Reviewer 2: ```suggestion
Note that the `--from-pagecache` option may be removed in future versions.
```

For more information on the block format, see xref:database-internals/store-formats.adoc[Store formats].
====

[NOTE]
====
In Neo4j 2025.01, the `--from-pagecache=<size>` option is replaced with a dual-named option: `--from-pagecache, --max-off-heap-memory=<size>`.
This change helps you to control off-heap memory rather than just allocate cache.

> Both options control the same underlying parameter, which determines the maximum amount of off-heap memory available for reading the source database during the copy process.

Author: ```suggestion
Both options control the same underlying parameter, which determines the maximum amount of off-heap memory available during the copy process, for both reading and writing.
```

[NOTE]
====
In Neo4j 2025.01, the `--from-pagecache=<size>` option is replaced with a dual-named option: `--from-pagecache, --max-off-heap-memory=<size>`.
This change helps you to control off-heap memory rather than just allocate cache.

> Both options control the same underlying parameter, which determines the maximum amount of off-heap memory available for reading the source database during the copy process.

Reviewer: It's not only about reading the source database, it's the total amount of off-heap memory the process will use, both for reading and for writing.

Finally, it is also important to consider that in almost all Cloud environments, the published IOPS value may not be the same as the actual value, or be able to continuously maintain the maximum possible IOPS.
The real processing time for this example _could_ be well above that estimation of 3 hours.

[[off-heap-memory-control]]
=== Improving the performance

You can improve the performance of the copy process by specifying the memory limit.
Neo4j 2025.01 introduces an option `--from-pagecache, --max-off-heap-memory=<size>` to replace the old `--from-pagecache=<size>` option.

The new option controls how much off-heap memory the copy process may use in addition to the heap size the JVM is given.
Values can be plain numbers, such as 10000000, or 20G for 20 gigabytes.
It can also be specified as a percentage of the available memory, for example 70%.

> Starting from 2025.01, to configure the off-heap memory value, you can use either the old name `--from-pagecache=<size>` or the new one `--max-off-heap-memory=<size>`.

Reviewer: Hmm why is the new one mentioned as both names here?

.Using `--from-pagecache=<size>`
[source, shell]
----
bin/neo4j-admin database copy neo4j copy --from-pagecache=95%
----

.Using `--max-off-heap-memory=<size>`
[source, shell]
----

> bin/neo4j-admin database copy neo4j copy --max-off-heap-memory=95%

Reviewer: Doesn't this look really weird? `--from-pagecache, --max-off-heap-memory=95%` What is `--from-pagecache` doing here?
